### ISSUE TITLE
Garage 0.8+ do not build on Darwin

### ISSUE THREAD

**Issue by RaitoBezarius** [NixOS/nixpkgs#208366]
### Steps To Reproduce
1. build `garage_0_8` introduced in this PR https://github.com/NixOS/nixpkgs/pull/196195 on Darwin (x86 or aarch64)
2. failed tests

### Build log
```

failures:

---- k2v::item::test_items_and_indices stdout ----
thread 'k2v::item::test_items_and_indices' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- admin::test_admin_bucket_perms stdout ----
thread 'admin::test_admin_bucket_perms' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- k2v::poll::test_poll stdout ----
thread 'k2v::poll::test_poll' panicked at 'Command exited with code 1: Could not assign garage node layout', src/garage/tests/common/ext/process.rs:46:31
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- k2v::batch::test_batch stdout ----
thread 'k2v::batch::test_batch' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- bucket::test_bucket_all stdout ----
thread 'bucket::test_bucket_all' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- k2v::item::test_item_return_format stdout ----
thread 'k2v::item::test_item_return_format' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- k2v::errorcodes::test_error_codes stdout ----
thread 'k2v::errorcodes::test_error_codes' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- k2v::simple::test_simple stdout ----
thread 'k2v::simple::test_simple' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::list::test_listmultipart stdout ----
thread 's3::list::test_listmultipart' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::list::test_listobjectsv1 stdout ----
thread 's3::list::test_listobjectsv1' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::list::test_listobjectsv2 stdout ----
thread 's3::list::test_listobjectsv2' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::multipart::test_uploadlistpart stdout ----
thread 's3::multipart::test_uploadlistpart' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::multipart::test_uploadpartcopy stdout ----
thread 's3::multipart::test_uploadpartcopy' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::objects::test_deleteobject stdout ----
thread 's3::objects::test_deleteobject' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::objects::test_getobject stdout ----
thread 's3::objects::test_getobject' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::objects::test_putobject stdout ----
thread 's3::objects::test_putobject' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::simple::test_simple stdout ----
thread 's3::simple::test_simple' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::streaming_signature::test_create_bucket_streaming stdout ----
thread 's3::streaming_signature::test_create_bucket_streaming' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::streaming_signature::test_put_website_streaming stdout ----
thread 's3::streaming_signature::test_put_website_streaming' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::streaming_signature::test_putobject_streaming stdout ----
thread 's3::streaming_signature::test_putobject_streaming' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::website::test_website_s3_api stdout ----
thread 's3::website::test_website_s3_api' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19

---- s3::website::test_website stdout ----
thread 's3::website::test_website' panicked at 'Once instance has previously been poisoned', src/garage/tests/common/garage.rs:215:19
```

### Additional context

I discussed with upstream and got this information:
> Once instance was poisonned indicatez that the garage server launched for the tests stopped unexpectedly
> Can be because the garage binary in $PATH isn't the correct one
> (The integration tests spawns a separate process that acts as a garage server, the garage server itself is not embedded in the task so it uses exec)
> If you want to debug more througklthoroughly, you can instrument the code of the integration testing framework in src/garage/tests

### Notify maintainers

@NickCao @0x4A6F @teutat3s

**Comment by wegank** [comment]
The tests seem to work fine on my machine (aarch64-darwin, with sandbox enabled):

```
running tests
Executing cargoCheckHook
++ cargo test -j 8 --release --features=k2v,kubernetes-discovery,bundled-libs,sled,metrics,telemetry-otlp,lmdb,sqlite --target aarch64-apple-darwin --frozen -- --test-threads=8
   Compiling num-integer v0.1.44
...
   Compiling garage v0.8.0 (/private/tmp/nix-build-garage-0.8.0.drv-0/source/src/garage)
    Finished release [optimized + debuginfo] target(s) in 6m 53s
     Running unittests main.rs (target/aarch64-apple-darwin/release/deps/garage-e571c83ab15e8949)

running 0 tests

test result: ok(B. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/lib.rs (target/aarch64-apple-darwin/release/deps/integration-9b7f6eeaea098d0a)

running 22 tests
test k2v::simple::test_simple ... ok(B
test k2v::poll::test_poll ... ok(B
test bucket::test_bucket_all ... ok(B
test admin::test_admin_bucket_perms ... ok(B
test k2v::errorcodes::test_error_codes ... ok(B
test k2v::item::test_item_return_format ... ok(B
test k2v::batch::test_batch ... ok(B
test s3::multipart::test_uploadlistpart ... ok(B
test s3::objects::test_getobject ... ok(B
test s3::list::test_listmultipart ... ok(B
test s3::simple::test_simple ... ok(B
test s3::streaming_signature::test_create_bucket_streaming ... ok(B
test s3::objects::test_putobject ... ok(B
test s3::streaming_signature::test_put_website_streaming ... ok(B
test s3::objects::test_deleteobject ... ok(B
test s3::streaming_signature::test_putobject_streaming ... ok(B
test s3::website::test_website ... ok(B
test s3::list::test_listobjectsv2 ... ok(B
test s3::list::test_listobjectsv1 ... ok(B
test s3::website::test_website_s3_api ... ok(B
test s3::multipart::test_uploadpartcopy ... ok(B
test k2v::item::test_items_and_indices ... ok(B

test result: ok(B. 22 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 19.15s

Finished cargoCheckHook
checkPhase completed in 7 minutes 13 seconds
installing
Executing cargoInstallHook
Finished cargoInstallHook
post-installation fixup
strip is /nix/store/fb4rdxr69sy1vsi98z3njxjzdq3rj3ah-clang-wrapper-11.1.0/bin/strip
stripping (with command strip and flags -S) in  /nix/store/fdb9ic1ivz5ni87g07zqixxpzlmybl6z-garage-0.8.0/bin
patching script interpreter paths in /nix/store/fdb9ic1ivz5ni87g07zqixxpzlmybl6z-garage-0.8.0
/nix/store/fdb9ic1ivz5ni87g07zqixxpzlmybl6z-garage-0.8.0
```

**Comment by RaitoBezarius** [comment]
I think this can be closed with #218843.
